core/src/test/scala/unit/kafka/server/ClientQuotaManagerTest.scala

```diff
@@ -418,6 +418,17 @@ class ClientQuotaManagerTest extends BaseClientQuotaManagerTest {
     }
   }
 
+  @Test
+  def testDelayedQueueSensorShouldShouldExistAfterInstantiationAndBeRemovedAfterShutdown(): Unit = {
+    val sensorName = Produce.toString + "-delayQueue"
+    val clientQuotaManager = new ClientQuotaManager(config, metrics, Produce, time, "")
+    var delayedQueueSensor = metrics.getSensor(sensorName)
+    assertNotNull(delayedQueueSensor, "delayed queue sensor should exist")
+    clientQuotaManager.shutdown()
+    delayedQueueSensor = metrics.getSensor(sensorName)
```

Comment:
Another validation we can perform is to ensure that
`assertTrue(metrics.metrics().isEmpty)` at `@AfterEach` of BaseClientQuotaManagerTest

Reply:
I looked onto this and apparently the metrics.metrics() List is never empty because this metric https://github.com/apache/kafka/blob/ca11a87e86e8e3c65043f747f35cae770b1efb7c/clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java#L180 isn't at the moment removed when we `Metrics#close()` is called.

Is that to be expected? If not, I can create a separate JIRA and address it separate from this PR.